zookeeper/tf/dataset.py

```diff
@@ -63,8 +64,14 @@ class TFDSDataset(Dataset):
     train_split: str
     validation_split: Optional[str] = None
 
+    should_cache: bool = True
+
+    @property
+    def should_cache(self):
+        return psutil.virtual_memory().available > (self.info.size_in_bytes * 1.05)
```

Comment:
This looks at the current available memory with 5% headroom. I am not sure if it should depend on the total memory of the system or if we should use a `@cached_property` so the flag doesn't change during a run.